tests/testthat/test-read_vpts.R

```diff
@@ -1,173 +1,226 @@
+
+  expect_warning(
+    read_vpts(file = vptsfile, radar = "KGBM", lat = 12, lon = 34, height = 1000, wavelength = "S"),
+    "read_stdout"
   )
-  expect_error(
-    read_vpts(vptsfile,
-      height = NA
-    ),
-    regexp = "height is not a number (a length one numeric vector).",
-    fixed = TRUE
+
+  # test read_stdout() txt without explicit extension
+  no_ext_file <- tempfile(pattern = "example_vpts")
+  file.copy(from = vptsfile, to = no_ext_file)
+
+  expect_warning(
+    read_vpts(file = no_ext_file, radar = "KBGM", lat = 12, lon = 34, height = 1000, wavelength = "S"),
+    "read_stdout"
   )
-  expect_error(
-    read_vpts(vptsfile,
-      radar = "KBGM",
-      height = seq(50)
-    ),
-    regexp = "height is not a number (a length one numeric vector).",
-    fixed = TRUE
+
+  expect_warning(
+    read_vpts(file = vptsfile, radar = "KBGM", lat = 12, lon = 34, height = 1000, wavelength = "S"),
+    "read_stdout"
   )
 
-  wavelength_msg <-
-    glue::glue(
-      "'wavelength' should be a single positive number",
-      ", or one of 'C' or 'S' for C-band and S-band radar, respectively."
+  # Test if outputs from both functions are equal but suppress warnings in tests
+  suppressWarnings(expect_equal(
+    read_vpts(files = vptsfile, radar = "radar", lat = 12, lon = 34, height = 1000),
+    read_stdout(file = vptsfile, radar = "radar", lat = 12, lon = 34, height = 1000, wavelength = "C", sep = "")
+  ))
+})
+
+test_that("read_vpts() returns error on explicit mixed extensions", {
+  # Prepare a vector of file paths with different extensions
+  files <- c("file1.csv", "file2.gz")
+  # Expect an error when calling read_vpts() with this input
+  expect_error(read_vpts(files), "`files` must all have the same extension.")
+})
+
+test_that("read_vpts() returns on error on inferred mixed extensions", {
+  # test an h5 file with a csv extension
+  example_h5 <- h5_files[1]
+  new_filename <- paste0(tools::file_path_sans_ext(basename(example_h5)), ".csv")
+  new_filepath <- file.path(other_dir, new_filename)
+  file.copy(example_h5, new_filepath)
+
+  expect_error(read_vpts(new_filepath), "does not match the guessed file type")
+})
+
+
+test_that("read_vpts() can read local vp hdf5 files", {
+  # Test for one file
+  {
+    result <- read_vpts_hdf5(h5_files[1])
+
+    expect_true(
+      length(result$datetime) == 1,
+      "Expected one vp object to be returned when reading one file."
     )
+    # Test if the output is a vpts object
+    expect_true(is.vpts(result))
+  }
+
+  # Test for multiple files
+  {
+    result <- read_vpts_hdf5(h5_files[1:2])
+
+    expect_true(
+      length(result$datetime) == 2,
+      "Expected two vp objects to be returned when reading two files."
+    )
+    # Test if the output is a vpts object
+    expect_true(is.vpts(result))
+  }
+})
+
+test_that("read_vpts() returns error on multiple radars in vp hdf5 files", {
+  skip("Not yet implemented")
```

Comment:
this one is skipped, but a critical test. can we add?

Reply:
read_vpts() will throws a warning instead of an error on multiple h5 radars.  This is because it invokes `bind_into_vpts()`

```r
> read_vpts(testfiles)
[[1]]
                   Regular time series of vertical profiles (class vpts)

           radar:  czbrd 
      # profiles:  3 
time range (UTC):  2023-06-01 00:00:00 - 2023-06-01 00:00:00 
   time step (s):  0 

[[2]]
                   Irregular time series of vertical profiles (class vpts)

           radar:  eehar 
      # profiles:  1 
time range (UTC):  2023-06-01 00:00:00 - 2023-06-01 00:00:00 
   time step (s):  min: NA     max:  NA 

Warning message:
In (function (...)  : Vertical profiles are not from a single radar.
```